modules/kubernetes-addons/datadog-operator/main.tf

```diff
@@ -0,0 +1,20 @@
+module "helm_addon" {
+  source            = "../helm-addon"
+  manage_via_gitops = var.manage_via_gitops
+  helm_config       = local.helm_config
+  irsa_config       = null
+  addon_context     = var.addon_context
+
+  depends_on = [kubernetes_namespace_v1.this]
+}
+
+resource "kubernetes_namespace_v1" "this" {
```

Comment:
we can remove this and instead use the `create_namespace` argument provided by the `helm_release` https://registry.terraform.io/providers/hashicorp/helm/latest/docs/resources/release#create_namespace